### ISSUE TITLE
IDISP003 should not warn when reassigning after dispose

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#153]
```cs
namespace Ns
{
    using System;
    using System.Threading;

    public abstract class C : IDisposable
    {
        private bool _disposed;
        private CancellationTokenSource _cancellationTokenSource;

        public void M()
        {
            try
            {
                _cancellationTokenSource?.Dispose();
                _cancellationTokenSource = new CancellationTokenSource();
            }
            finally
            {
                _cancellationTokenSource?.Dispose();
                _cancellationTokenSource = null;
            }
        }

        public void Dispose()
        {
            Dispose(true);
            GC.SuppressFinalize(this);
        }

        protected virtual void Dispose(bool disposing)
        {
            if (_disposed)
            {
                return;
            }

            _disposed = true;
            if (disposing)
            {
                _cancellationTokenSource?.Dispose();
            }
        }
    }
}
```
